### PR TITLE
feat: l2g feature to indicate if gene is protein-coding or not

### PR DESCRIPTION
**Pull request by xyg123**
## ✨ Context

Adding a feature to tell l2g when a gene is protein coding or not
https://github.com/opentargets/issues/issues/3509

## 🛠 What does this PR implement

A new feature that is either 0 or 1, for each studylocus-to-gene mapping, 1 when the gene is protein-coding, 0 otherwise.

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/dataset/l2g_features/other.py [hunk 1]
```diff
@@ -81,6 +81,51 @@ def common_genecount_feature_logic(
     )
 
 
+def common_protein_coding_feature_logic(
```
Comment (ireneisdoomed) [inline src/gentropy/dataset/l2g_features/other.py:84]: Since this is only going to be one feature, I don't see the need in having a `common` function.

**Comment by addramir** [comment]
Looks good to me. @ireneisdoomed please have a look.
BTW why is it named isProteinCoding1mb? Why 1mb?

**Review by ireneisdoomed — APPROVED** [review]
LGTM

**Comment by xyg123** [comment]
> Looks good to me. @ireneisdoomed please have a look. BTW why is it named isProteinCoding1mb? Why 1mb?

The feature is generated for each study locus, and it’s set to 1mb because, by default, VEP and other related annotation tools use this range to define potential effector genes.

**Comment by ireneisdoomed** [comment]
@xyg123 I agree with @addramir in that the window only matters to build the list of variant/gene pairs. If the gene is protein coding or not is not dependent of the window.

**Pull request merged**